### ISSUE TITLE
Assertion error when using existing `log_group` for `awsx.ecs.FargateTaskDefinition`

### ISSUE THREAD

**Issue by WillDaSilva** [pulumi/pulumi-awsx#929]
### What happened?

Using Python I wanted to create a `awsx.ecs.FargateTaskDefinition` that used a previously created log group. I tried to use the `log_group` parameter for this, which has the type `Optional[pulumi.InputType['_awsx.DefaultLogGroupArgs']]`. 

### Steps to reproduce

A `awsx.ecs.FargateTaskDefinition` instance was created with the log group set as follows:

```python
awsx.ecs.FargateTaskDefinition(
    resource_name=f"{resource_key}-orchestrator-task",
    log_group=awsx.awsx.DefaultLogGroupArgs(
        existing=awsx.awsx.ExistingLogGroupArgs(arn=log_group.arn)
    ),
    container=awsx.ecs.TaskDefinitionContainerDefinitionArgs(
        command=command,
        image=image.image_uri,
        cpu=256,
        memory=512,
        essential=True,
    ),
)
```

Where `log_group.arn` is the ARN of a previously created log group.

### Expected Behavior

The previously created log group should be used by tasks created by the task definition. No errors should occur during `pulumi up`.

### Actual Behavior

The following error appears in the diagnostics section after running `pulumi up`.

```
    error: Program failed with an unhandled exception:
    Traceback (most recent call last):
      File "/home/will/.pulumi/bin/pulumi-language-python-exec", line 179, in <module>
        loop.run_until_complete(coro)
      File "/home/will/.pyenv/versions/3.10.7/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
        return future.result()
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 126, in run_in_stack
        await run_pulumi_func(lambda: Stack(func))
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 51, in run_pulumi_func
        await wait_for_rpcs()
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 110, in wait_for_rpcs
        raise exception
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/output.py", line 168, in run
        is_known = await self._is_known
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/output.py", line 99, in is_value_known
        return await is_known and not contains_unknowns(await future)
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/resource.py", line 653, in do_register
        rpc.resolve_outputs(
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 1049, in resolve_outputs
        all_properties[translated_key] = translate_output_properties(
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 882, in translate_output_properties
        raise AssertionError(
    AssertionError: Unexpected type; expected a value of type `<class 'pulumi_aws.cloudwatch.log_group.LogGroup'>` but got a value of type `<class 'dict'>` at resource `ABCD-orchestrator-task`, property `log_group`: {'existing': {'arn': <pulumi.output.Output object at 0x7f5bc865b460>}}
```

### Output of `pulumi about`
```
CLI          
Version      3.42.0
Go Version   go1.19.1
Go Compiler  gc

Plugins
NAME     VERSION
aws      5.16.0
awsx     1.0.0-beta.11
docker   3.4.1
python   unknown

Host     
OS       linuxmint
Version  20.3
Arch     x86_64

This project is written in python: executable='/home/will/.pyenv/shims/python3' version='3.10.7
'

Current Stack: dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend        
Name           desktop
URL            file://~
User           will
Organizations  

Pulumi locates its logs in /tmp by default
```

### Additional context

The documentation for the `log_group` parameter of `FargateTaskDefinition` is:

> A set of volume blocks that containers in your task may use.

Which seems wrong.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
Thank you for the detailed writeup @WillDaSilva 

Just to confirm - this works correctly before adding the `log_group > existing > arn` property - then just adding this causes the error you're seeing?

A few questions to help us narrow down the issue:
- Does `pulumi preview` throw the same error?
- Does the resource get created correctly before it fails with the error?

We'll need to dig into the exact source of this error - I'll need to confirm if it affect languages other than python too.

**Comment by WillDaSilva** [comment]
> Just to confirm - this works correctly before adding the log_group > existing > arn property - then just adding this causes the error you're seeing?

Correct.

> Does `pulumi preview` throw the same error?

No, there is no error when running `pulumi preview`.

> Does the resource get created correctly before it fails with the error?

Yes, but when the error occurs a different resource in the stack fails to be created. For context, this log group is provided to an `awsx.ecs.FargateTaskDefinition`, which is then used to create an `aws.cloudwatch.EventTarget`. The event target is silently not created when this error occurs.

Thanks for looking into this @danielrbradley! Please let me know if you need any additional info.

**Comment by WillDaSilva** [comment]
Any progress on this @danielrbradley? This and #933 have become more urgent for us. Any pointers for how to fix this would be appreciated.

**Comment by WillDaSilva** [comment]
When the `log_group` parameter of `awsx.ecs.FargateTaskDefinition` is set to `awsx.awsx.ExistingLogGroupArgs(arn=log_group.arn)` instead of `awsx.awsx.DefaultLogGroupArgs(existing=awsx.awsx.ExistingLogGroupArgs(arn=log_group.arn))` no error occurs during `pulumi up`. Unfortunately, despite the lack of an error, this is not a solution. The specified log group is not used by `awsx.ecs.FargateTaskDefinition`. Instead it creates a new log group as if the `log_group` parameter had been set to `None`.

**Comment by danielrbradley** [comment]
Sorry, for the slow progress on this issue. 

Having reproduced the issue myself, this stems from a core SDK bug where it appears to default to the input value if an output value isn't provided. I've logged https://github.com/pulumi/pulumi/issues/11416 to find a resolution.

**Comment by WillDaSilva** [comment]
@danielrbradley Using Pulumi version 3.47.2, this has not been fixed. The error is the same as before.

**Comment by danielrbradley** [comment]
This is an internal dependency bump - you'll need to upgrade awsx to V1.0.0

**Comment by WillDaSilva** [comment]
After upgrading `pulumi-awsx` to `v1.0.0` I get the following error from the above code:

```
AttributeError: 'NoneType' object has no attribute 'arn'
```

If I restructure it like so:

```python
awsx.ecs.FargateTaskDefinition(
    resource_name=f"{resource_key}-orchestrator-task",
    log_group=log_group.arn.apply(
        lambda arn: awsx.awsx.DefaultLogGroupArgs(
            existing=awsx.awsx.ExistingLogGroupArgs(arn=arn)
        )
    ),
    container=awsx.ecs.TaskDefinitionContainerDefinitionArgs(
        command=command,
        image=image.image_uri,
        cpu=256,
        memory=512,
        essential=True,
    ),
)
```

Then I get the following error:

```
AttributeError: 'dict' object has no attribute 'arn'
```

@danielrbradley Was the code provided in this issue ran before closing this issue? If so, how did you run it successfully?

**Comment by justinvp** [comment]
@WillDaSilva, I'm unable to reproduce the error you're seeing. The following program runs without any errors:

```python
import pulumi_aws as aws
import pulumi_awsx as awsx

log_group = aws.cloudwatch.LogGroup("myloggroup")

awsx.ecs.FargateTaskDefinition(
    resource_name="task_def",
    log_group=awsx.awsx.DefaultLogGroupArgs(
        existing=awsx.awsx.ExistingLogGroupArgs(arn=log_group.arn)
    ),
    container=awsx.ecs.TaskDefinitionContainerDefinitionArgs(
        image="nginx:latest",
        cpu=256,
        memory=512,
        essential=True,
    ),
)
```

These are the packages I have installed, notably `pulumi 3.48.0` and `pulumi-awsx 1.0.0`

```
$ venv/bin/python -m pip list
Package       Version
------------- -------
Arpeggio      2.0.0
attrs         22.1.0
dill          0.3.6
grpcio        1.47.0
parver        0.4
pip           22.3.1
protobuf      4.21.9
pulumi        3.48.0
pulumi-aws    5.21.1
pulumi-awsx   1.0.0
pulumi-docker 3.6.1
PyYAML        6.0
semver        2.13.0
setuptools    65.6.3
six           1.16.0
wheel         0.38.4
```

**Comment by WillDaSilva** [comment]
@justinvp I have found the issue. After using the `log_group` parameter as you've done in your example, the `log_group` attribute of the `awsx.ecs.FargateTaskDefinition` instance is `None`, but its value ought to be the provided `log_group` object. This is a different issue than #929, so I'll close #929, and open a new issue.
